### PR TITLE
Fix the `verify-bundles` freeze helper script.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This PR fixes the `verify-bundles` script by:
1. Changing the extensions for the ES builds from `js` to `mjs` for:
    1. `@handsontable/react`
    2. `@handsontable/vue` 
    3. `@handsontable/vue3`
2. Added entries for the UMD and minified UMD builds to the `exports` section of `package.json` for:
    1. `@handsontable/react`
    2. `@handsontable/vue` 
    3. `@handsontable/vue3`
3. Adding the JSON type assertions to the dynamic imports in the `verify-bundles` script.

I haven't changed the `exports` entry in `@handsontable/angular` as currently the only export it uses is targeted at its `umd` build, which should probably be reviewed in a separate task.

### How has this been tested?
Run the `verify-bundles` script by running `node ./scripts/verify-bundles.mjs` from the root directory.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9891 

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: scripts/verify-bundles.mjs [hunk 1]
```diff
@@ -39,7 +39,10 @@ async function verifyBundles() {
       className: 'HotTable'
     }
   };
-  const { default: mainPackageJson } = await import('../package.json');
+  const { default: mainPackageJson } = await import('../package.json', {
+    assert: { type: 'json' }
+    }
```
Comment (budnix) [inline scripts/verify-bundles.mjs:44]: Weird formatting 🤔

**Review by budnix — CHANGES_REQUESTED** [review]
Everything looks good 👏 

I have one suggestion that bothers me. The latest Node version that we use supports top-level await syntax, the script does only one job, WDYT about shifting the whole code to one tab left and extracting the logic out of the `verifyBundles` function? Then the function can be removed. I mean:
```js
async function verifyBundles() {
  // the script body
}
verifyBundles();
```
to
```js
// the script body
```

File: scripts/verify-bundles.mjs (hunk 1 above)
Comment (jansiegel) [reply scripts/verify-bundles.mjs:44]: I agree. The linter forced me to do this 🙄 https://github.com/handsontable/handsontable/pull/10068/commits/1326ed95b00e1787d3945724f66525915d5ee2ba

Comment (budnix) [reply scripts/verify-bundles.mjs:44]: What about a one-liner? 
`const { default: mainPackageJson } = await import('../package.json', { assert: { type: 'json' }});`
or 
```js
const { 
  default: mainPackageJson
} = await import('../package.json', { assert: { type: 'json' }});
```

**Review by budnix — APPROVED** [review]
Looks good 👌

**Pull request merged**